core/embed/rust/src/ui/model_tr/screens.rs

```diff
@@ -1,29 +1,20 @@
-#[cfg(feature = "micropython")]
-use crate::micropython::buffer::StrBuffer;
-use crate::ui::{
-    component::base::Component, constant::screen, display, model_tr::component::WelcomeScreen,
+use crate::{
+    strutil::TString,
+    ui::{
+        component::base::Component, constant::screen, display, model_tr::component::WelcomeScreen,
+    },
 };
 
 use super::{component::ErrorScreen, constant};
 
-#[cfg(not(feature = "micropython"))]
-// SAFETY: Actually safe but see below
-unsafe fn get_str(text: &str) -> &str {
-    text
-}
-#[cfg(feature = "micropython")]
-// SAFETY: The caller is responsible for ensuring that the StrBuffer does not
-// escape the lifetime of the original &str.
-unsafe fn get_str(text: &str) -> StrBuffer {
-    unsafe { StrBuffer::from_ptr_and_len(text.as_ptr(), text.len()) }
+fn get_str(text: &str) -> TString {
+    TString::Str(text)
 }
 
 pub fn screen_fatal_error(title: &str, msg: &str, footer: &str) {
-    // SAFETY: these will get placed into `frame` which does not outlive this
-    // function
-    let title = unsafe { get_str(title) };
-    let msg = unsafe { get_str(msg) };
-    let footer = unsafe { get_str(footer) };
+    let title = get_str(title);
```

Comment:
copying a comment i've seen outside the review context: we don't need `get_str()` anymore, we can just call `title.into()` etc.

Reply:
d0243af2e2ecd705cf927cfa3220d61d107fc09b